net/tsdial/tsdial.go

```diff
@@ -75,7 +77,8 @@ type Dialer struct {
 	exitDNSDoHBase   string                 // non-empty if DoH-proxying exit node in use; base URL+path (without '?')
 	dnsCache         *dnscache.MessageCache // nil until first non-empty SetExitDNSDoH
 	nextSysConnID    int
-	activeSysConns   map[int]net.Conn // active connections not yet closed
+	activeSysConns   map[int]net.Conn                 // active connections not yet closed
+	routes           atomic.Pointer[bart.Table[bool]] // or nil if UserDial should not use routes. `true` indicates routes that point into the Tailscale interface
```

Comment:
nit: as this is no longer protected by `mu` move it before the block so that its clearer that its not protected.